src/Ryujinx.Headless.SDL2/Ryujinx.Headless.SDL2.csproj

```
@@ -29,6 +34,7 @@
 
   <ItemGroup>
     <PackageReference Include="CommandLineParser" />
+    <PackageReference Include="Ryujinx.Graphics.Vulkan.Dependencies.MoltenVK" Condition="'$(RuntimeIdentifier)' != 'linux-x64' AND '$(RuntimeIdentifier)' != 'win10-x64'" />
```

```suggestion
    <PackageReference Include="Ryujinx.Graphics.Vulkan.Dependencies.MoltenVK" Condition="'$(RuntimeIdentifier)' == 'osx-x64' OR '$(RuntimeIdentifier)' == 'osx-arm64' OR $([MSBuild]::IsOSPlatform('OSX'))" />
```
How about something like this? Maybe this could also help with #5049? (cc @marysaka)

---

I would prefer to keep the solution the same as we have on other projects for now @TSRBerry.
I think we should unify this kind of checks in some helper file to avoid duplicating it everywhere.